### PR TITLE
Add tests for ProxyCommand APIs to improve coverage

### PR DESCRIPTION
**Pull request by daxian-dbw**
Address #4164
Examine the code in ProxyCommand.cs to find out specific missing areas.

### CODE REVIEW

File: test/powershell/engine/Api/ProxyCommand.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        $helpObj.Synopsis | Should Be $newHelpObj.Synopsis
+        $oldDespText = GetSectionText $helpObj.description
+        $newDespText = GetSectionText $newHelpObj.description
+        $oldDespText | Should Be $newDespText
+
+        $oldParameters = @($helpObj.parameters.parameter)
+        $newParameters = @($newHelpObj.parameters.parameter)
+        $oldParameters.Length | Should Be $newParameters.Length
+        for ($i = 0; $i -lt $oldParameters.Length; $i++) {
+            $oldParameters[$i].name | Should Be $newParameters[$i].name
```
Comment (iSazonov) [inline test/powershell/engine/Api/ProxyCommand.Tests.ps1]: Do we really want report every value? Can we use `Compare-Object` to be fast?

Comment (daxian-dbw) [reply test/powershell/engine/Api/ProxyCommand.Tests.ps1]: Replaced with `$oldParameters.name -join "," | Should Be ($newParameters.name -join ",")`

File: test/powershell/engine/Api/ProxyCommand.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+                [Parameter(ParameterSetName="Id", Mandatory=$true)]
+                [ValidateRange(1,5)]
+                [int] $Id,
+
+                [Parameter(ValueFromPipeline)]
+                [string] $Message
+            )
+
+            DynamicParam {
```
Comment (anmenaga) [inline test/powershell/engine/Api/ProxyCommand.Tests.ps1]: Why DynamicParams are needed here?

Comment (anmenaga) [reply test/powershell/engine/Api/ProxyCommand.Tests.ps1]: Nevermind; I see it in the nd of the file.

**Review by anmenaga — APPROVED** [review]
LGTM

**Comment by daxian-dbw** [comment]
Thanks @anmenaga!

**Pull request merged**